src/main/scala/ch/epfl/lamp/SbtCoursera.scala

```diff
@@ -281,80 +276,36 @@ object SbtCourseraPlugin extends AutoPlugin {
     val gradingCourseId = SettingKey[String]("gradingCourseId")
 
     /**
-     * The api key to access non-public api parts on coursera. This key is secret! It's defined in
-     * 'submission/settings.sbt', which is not part of the handout.
-     *
-     * Default value 'apiKey' to make the handout sbt project work
-     *  - In the handout, apiKey needs to be defined, otherwise the build doesn't compile
-     *  - When correcting, we define 'apiKey' in the 'submission/sectrets.sbt' file
-     *  - The value in the .sbt file will take precedence when correcting (settings in .sbt take
-     *    precedence over those in .scala)
-     */
-    val apiKey = SettingKey[String]("apiKey")
-
-    /**
-     * **********************************************************
-     * GRADING INITIALIZATION
-     */
-
-    val initGrading = TaskKey[Unit]("initGrading")
-    lazy val initGradingSetting = initGrading <<= (clean, baseDirectory, sourceDirectory) map { (_, baseDir, submissionSrcDir) =>
-      deleteFiles(submissionSrcDir, baseDir)
-      GradingFeedback.initialize()
-      RecordingLogger.clear()
-    }
-
-    def deleteFiles(submissionSrcDir: File, baseDir: File) {
-      // don't delete anything in offline mode, useful for us when hacking testing / stylechecking
-      if (!Settings.offlineMode) {
-        IO.delete(submissionSrcDir)
-        IO.delete(baseDir / Settings.submissionJarFileName)
-        IO.delete(baseDir / Settings.testResultsFileName)
-      }
-    }
-
-    /**
      * **********************************************************
      * DOWNLOADING AND EXTRACTING SUBMISSION
      */
 
     val getSubmission = TaskKey[Unit]("getSubmission")
-    val getSubmissionSetting = getSubmission <<= (initGrading, baseDirectory, scalaSource in Compile) map { (_, baseDir, scalaSrcDir) =>
-      readAndUnpackSubmission(baseDir, scalaSrcDir)
-    }
-
-    def readAndUnpackSubmission(baseDir: File, targetSourceDir: File) {
+    val getSubmissionSetting = getSubmission := {
+      GradingFeedback.initialize()
+      RecordingLogger.clear()
       try {
-        val jsonFile = baseDir / Settings.submissionJsonFileName
-        val targetJar = baseDir / Settings.submissionJarFileName
```

Comment:
`Settings.submissionJarFileName` should be dead code now but I do not see the deletion of it in the PR.